src/ipc/ipc4/helper.c

```diff
@@ -235,15 +236,21 @@ int ipc_pipeline_free(struct ipc *ipc, uint32_t comp_id)
 static struct comp_buffer *ipc4_create_buffer(struct comp_dev *src, struct comp_dev *sink,
 					      uint32_t src_queue, uint32_t dst_queue)
 {
-	struct ipc4_base_module_cfg *src_cfg;
+	const struct comp_driver *drv = src->drv;
+	struct ipc4_base_module_cfg src_cfg;
 	struct comp_buffer *buffer = NULL;
 	struct sof_ipc_buffer ipc_buf;
-	int buf_size;
+	int buf_size, ret;
 
-	src_cfg = (struct ipc4_base_module_cfg *)comp_get_drvdata(src);
+	ret = drv->ops.get_attribute(src, COMP_ATTR_BASE_CONFIG, &src_cfg);
```

Comment:
Why `get_attribute` doesn't return just a pointer to ipc4_base_module_cfg?

```
struct ipc4_base_module_cfg *src_cfg;
ret = drv->ops.get_attribute(src, COMP_ATTR_BASE_CONFIG, &src_cfg);
buf_size = src_cfg->obs * 2;
```

Reply:
@softwarecki I dont think we should return a pointer here. The base config is part of the module's private data and common core code should not have access to it.

Reply:
It's always a trade off between performance and security. The `ipc4_comp_get_base_module_cfg` function may just return a const pointer. This problem will come back when somebody will need access to a higher level of the `ipc4_base_module_cfg` structure, eg. `ipc4_copier_module_cfg`, `ipc4_up_down_mixer_module_cfg` etc.

Reply:
@softwarecki the base config is needed during widget binding and during trigger AFAICT. so performance shouldn't be a huge problem as it isnt it?

Reply:
So if I look forwards, we will have a clear privilege separation between base firmware and modules (protected by Zephyr). In this case we need to pass a copy and this will be needed in other places too. Our eventual module <-> baseFW data passing will looks like Linux kernel <-> userpace data passing.